steering_controllers/include/steering_controllers/steering_odometry.hpp

```diff
@@ -0,0 +1,209 @@
+     */
+  double get_x() const { return x_; }
+
+  /**
+     * \brief y position getter
+     * \return y position [m]
+     */
+  double get_y() const { return y_; }
+
+  /**
+     * \brief linear velocity getter
+     * \return linear velocity [m/s]
+     */
+  double get_linear() const { return linear_; }
+
+  /**
+     * \brief angular velocity getter
+     * \return angular velocity [rad/s]
+     */
+  double get_angular() const { return angular_; }
+
+  /**
+     * \brief Sets the wheel parameters: radius, separation and wheelbase
+     */
+  void set_wheel_params(
+    double wheel_radius, double wheel_separation_h = 0.0, double wheelbase = 0.0);
+
+  /**
+     * \brief Velocity rolling window size setter
+     * \param velocity_rolling_window_size Velocity rolling window size
+     */
+  void set_velocity_rolling_window_size(size_t velocity_rolling_window_size);
+
+  /**
+     * \brief TODO
+     * \param Vx  TODO
+     * \param theta_dot TODO
+     */
+  std::tuple<double, double> twist_to_ackermann(double Vx, double theta_dot);
+
+  /**
+     *  \brief Reset poses, heading, and accumulators
+     */
+  void reset_odometry();
+
+private:
+  /**
+     * \brief Uses precomputed linear and angular velocities to compute dometry and update accumulators
+     * \param linear  Linear  velocity   [m] (linear  displacement, i.e. m/s * dt) computed by previous odometry method
+     * \param angular Angular velocity [rad] (angular displacement, i.e. m/s * dt) computed by previous odometry method
+     */
+  bool update_odometry(const double linear_velocity, const double angular, const double dt);
+
+  /**
+     * \brief Integrates the velocities (linear and angular) using 2nd order Runge-Kutta
+     * \param linear  Linear  velocity   [m] (linear  displacement, i.e. m/s * dt) computed by encoders
+     * \param angular Angular velocity [rad] (angular displacement, i.e. m/s * dt) computed by encoders
+     */
+  void integrate_runge_kutta_2(double linear, double angular);
+
+  /**
+     * \brief Integrates the velocities (linear and angular) using exact method
+     * \param linear  Linear  velocity   [m] (linear  displacement, i.e. m/s * dt) computed by encoders
+     * \param angular Angular velocity [rad] (angular displacement, i.e. m/s * dt) computed by encoders
+     */
+  void integrate_exact(double linear, double angular);
+
+  /**
+    * \brief TODO
+    * \param Vx  TODO
+    * \param theta_dot TODO
+    */
+  double convert_trans_rot_vel_to_steering_angle(double Vx, double theta_dot);
+
+  /**
+     *  \brief Reset linear and angular accumulators
+     */
+  void reset_accumulators();
+
+  /// Current timestamp:
+  rclcpp::Time timestamp_;
+
+  /// Current pose:
+  double x_;        //   [m]
+  double y_;        //   [m]
+  double heading_;  // [rad]
+
+  /// Current velocity:
+  double linear_;   //   [m/s]
+  double angular_;  // [rad/s]
+
+  double wheel_separation_;
+  double wheelbase_;
+  double wheel_radius_;
```

Comment:
can we allow for different sized wheel radii? at least front and rear axle, but I wonder if there is an even more portable solution.
I am not sure, but how might different front/rear wheel_separations also play into the downstream odom calculations?

Reply:
I have enabled different sized radii and track lengths using name conventions as here: https://en.wikipedia.org/wiki/Axle_track#/media/File:Wheelbase_and_Track.png

Also, rear steering is now enabled by setting `front_steering` parameter to false. I still need to check if different wheel tracks influence odometry equations.